src/main/java/step4/domain/car/CarList.java

```diff
@@ -0,0 +1,62 @@
+package step4.domain.car;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CarList {
+	List<Car> cars;
+	int movedCount;
+
+	public CarList(List<Car> cars) {
+		this.cars = cars;
+		this.movedCount = 0;
+	}
+
+	public static CarList from(String[] nameList) {
+		List<Car> cars = new ArrayList<>();
+		for (String name : nameList) {
+			cars.add(new Car(name));
+		}
+		return new CarList(cars);
+	}
+
+	private static int getMaxPosition(Car car, int maxValue) {
+		if (maxValue < car.getCurrentPosition()) {
+			maxValue = car.getCurrentPosition();
+		}
+		return maxValue;
+	}
+
+	public void moveCarList() {
+		for (Car car : cars) {
+			car.move(car.getMoveStrategy().movable());
+		}
+		this.movedCount++;
+	}
+
+	public List<Car> getList() {
+		return cars;
+	}
+
+	public int getSize() {
+		return cars.size();
+	}
+
+	public List<Car> getWinner() {
+		List<Car> winnerList = new ArrayList<>();
+		int maxPosition = 0;
+		for (Car car : cars) {
+			maxPosition = getMaxPosition(car, maxPosition);
+		}
+		for (Car car : cars) {
+			if (car.getCurrentPosition() == maxPosition) {
+				winnerList.add(car); // 이거 줄일수가 없어요...
+			}
+		}
```

Comment:
여러 방법이 있는데, 가장 좋은 방법은 행위를 분리하는 것이죠. 
```suggestion
public List<Car> getWinner() {
  int maxPosition = findMaxPosition(cars);
  return filterWinnerCars(cars, maxPosition);
}

private int findMaxPosition(List<Car> cars) {
  int maxPosition = 0;
  for (Car car : cars) {
    maxPosition = getMaxPosition(car, maxPosition);
  }
  return maxPosition;
}

private List<Car> filterWinnerCars(List<Car> cars, int maxPosition) {
  return cars.stream()
    .filter(car -> car.getCurrentPosition() == maxPosition)
    .collect(Collectors.toList());
}

```

Reply:
단어가 익숙하지 않아서 그런데 행위를 분리한다는 것이 
한가지 함수에는 한가지 일만 하게 하는거 얘기하시는거죠?
이것도 익숙해지도록 노력해보겠습니다.

Reply:
한가지 일만 하는 함수는 순수함수(pure function)이라 하는데요, 행위를 분리한다는게 모든 함수들이 순수함수로 가는 방향이라는 부분에선 일정부분 맞는말이나, 행위를 분리할 때 무조건 순수함수로 만들어야 한다. 는 말은 또 아닙니다. 

- 함수내의 추상화의 수준을 맞춰라 
- 함수내의 들여쓰기 수준을 2 depth 이하로 맞춰라

우선은 두 가지 컨벤션을 지켜보도록 합시다. 실무에선 다 지키기 힘들지만, 학습과정이니만큼 가치가 있습니다 ㅎ